libevmasm/ConstantOptimiser.cpp

```diff
@@ -253,7 +253,47 @@ AssemblyItems ComputeMethod::findRepresentation(u256 const& _value)
 	if (_value < 0x10000)
 		// Very small value, not worth computing
 		return AssemblyItems{_value};
-	else if (numberEncodingSize(~_value) < numberEncodingSize(_value))
+
+	// check for masks first
+	unsigned lowZeros = 0;
+	unsigned highOnes = 0;
+	for (; ((_value >> lowZeros) & 1) == 0 && lowZeros < 256; lowZeros++) {}
+	for (; ((_value >> (lowZeros + highOnes)) & 1) == 1 && highOnes < 256; highOnes++) {}
```

Comment:
Some remarks on readability of this.

One thing that would have helped me grasp what's happening much quicker the first time I saw this code would be an example what the "high ones" and "low zeros" are. From the name I actually expected the ones to be the highest bits.

I'd also generally use `while` and less obvious parentheses to make it a little less dense:

```suggestion
	//                                        high ones      low zeros
	//                                       |----------||--------------|
	// 0x000000000000000000000000000000000000ffffffffffff0000000000000000
	unsigned lowZeros = 0;
	unsigned highOnes = 0;
	while ((_value >> lowZeros) & 1 == 0 && lowZeros < 256)
		++lowZeros;
	while ((_value >> (lowZeros + highOnes)) & 1 == 1 && highOnes < 256)
		++highOnes;

```

For readability I'd also appreciate some more separation between each of methods you try below (e.g. a clear comment starting it and a blank like after a block with `return`; or even just making them independent helper methods with nice names).

Reply:
The parenthesis were necessary (the suggested code does not compile). changed the rest as suggested.